src/Drivers/hiopbbpy/BODriver.py

```diff
@@ -17,34 +21,109 @@
 from hiopbbpy.problems import BraninProblem
 
 
-### parameters
-n_samples = 5 # number of the initial samples to train GP
-theta = 1.e-2 # hyperparameter for GP kernel
+# Get user input for the number of repetitions from command-line arguments
+if len(sys.argv) != 2 or int(sys.argv[1]) < 0:
+    num_repeat = 1
+else:
+    num_repeat = int(sys.argv[1])
 
-nx = 2 # dimension of the problem
+### parameters
+n_samples = 5  # number of the initial samples to train GP
+theta = 1.e-2  # hyperparameter for GP kernel
+nx = 2         # dimension of the problem
 xlimits = np.array([[-5, 5], [-5, 5]]) # bounds on optimization variable
 
-#problem = LpNormProblem(nx, xlimits)
-problem = BraninProblem()
-
-print(problem.name, " problem")
-
-### initial training set
-x_train = problem.sample(n_samples)
-y_train = problem.evaluate(x_train)
-
-# Define the GP surrogate model
-gp_model = smtKRG(theta, xlimits, nx)
-gp_model.train(x_train, y_train)
-
-acquisition_types = ["LCB", "EI"]
-for acquisition_type in acquisition_types:
-    print("acquisition type: ", acquisition_type)
-    
-    # Instantiate and run Bayesian Optimization
-    bo = BOAlgorithm(gp_model, x_train, y_train, acquisition_type = acquisition_type) #EI or LCB
-    bo.optimize(problem)
-    
-    # Retrieve optimal point
-    x_opt, y_opt = bo.getOptimalPoint()
-    print()
+### saved solutions --- from 1000 repetitions
+saved_min_obj = {"LpNorm": {"LCB": 0.0007586314501994839, "EI": 0.002094016049616341}, "Branin": {"LCB": 0.3979820338569908, "EI": 0.39789916461969455}}
+saved_mean_obj = {"LpNorm": {"LCB": 0.018774638321851504, "EI": 0.11583915178648867}, "Branin": {"LCB": 0.5079001079219421, "EI": 0.4377466109837465}}
+saved_max_obj = {"LpNorm": {"LCB": 0.0755173754382861, "EI": 0.4175676394969743}, "Branin": {"LCB": 1.107240543567082, "EI": 0.7522382699410031}}
+saved_yopt = np.load("yopt_20iter_1000run.npy",allow_pickle=True).item()
```

Comment:
This file is not saved to this repo and so if I were to clone the repo and try to run this file it would fail due to not being able to find the `yopt_20iter_1000run.npy`. Perhaps though, it doesn't need to be saved. Offline you can determine the 5th and 95th percentile.

Reply:
I apologize, I see the file now. Disregard my previous comment about the `.npy` file not being present.